xbmc/addons/Addon.h

```diff
@@ -36,8 +36,8 @@ namespace ADDON
 
   const char* const ORIGIN_SYSTEM = "b6a50484-93a0-4afb-a01c-8d17e059feda";
 
-void OnEnabled(const std::string& id);
-void OnDisabled(const std::string& id);
+void OnEnabled(const AddonPtr& addon);
+void OnDisabled(const AddonPtr& addon);
 void OnPreInstall(const AddonPtr& addon);
 void OnPostInstall(const AddonPtr& addon, bool update, bool modal);
 void OnPreUnInstall(const AddonPtr& addon);
```

Comment:
The reason these static functions were added in the first place is because there are multiple different versions of add-ons (e.g the one in the repository, and ones that were loaded as a different type). And the result is that you if you call it on the "wrong" instance you will end up calling the wrong hook.

Better to just fix the remaining implementations of these On* methods than polish this turd.